File: tree/balance.py
class Node:
    def __init__ (self,key):
        self.left=None
        self.right=None
        self.data=key
def hieght(root):
    if root is None:
        return 0
    return 1+max(hieght(root.left),hieght(root.right));
def isBalence(root):
    if root is None:
        return 1
    lh=hieght(root.left)
    rh=hieght(root.right)
    if abs(lh-rh)<=1 and isBalence(root.left)and isBalence(root.right):
        return 1
    return 0
root=Node(1)
root.left=Node(2)
root.right=Node(3)
root.left.left=Node(4)
if isBalence(root):
    print("Tree is Balenced")
else:
    print("Not Balenced")
    

File: tree/Balance.java
class Node
  {
   int data;
  Node left,right;
   Node(int item)
    {
       data=item;
      left=right=null;
     }    
}

class Balance
{
  static Node root;
  boolean isBalance(Node node)
      {
        int lh,rh;
           if(node==null)
              return true;
            lh=hieght(node.left);
            rh=hieght(node.right);
               if(Math.abs(lh-rh<=1)&& isBalance(node.left)&&(node.right))
           return false;
       }
int max(int x,int y)
{
     return (x>y)?x:y;
}
int hieght(Node node)
  {
    if(node==null)
       return 1;
      return 1+max(hieght(node.left),hieght(node.right));
 }
   public static void main(String arg[])
    {
      Balance tree=new Balance();
      tree.root=new Node(1);
      tree.root.left=new Node(2);
      tree.root.left.right=new Node(3);
      tree.root.left.left=new Node(4);
      tree.root.right=new Node(5);
     tree.root.right.left=new Node(6);
     tree.root.left.left.left=new Node(7);
 if(tree.isBalance(root))
   System.out.println("tree is balanced..");
else
  System.out.println("Not balanced");
     }
}